vk_video_decoder/libs/NvVideoParser/src/VulkanAV1GlobalMotionDec.cpp

```diff
@@ -166,9 +166,9 @@ uint16_t VulkanAV1Decoder::Read_primitive_refsubexpfin(uint16_t n, uint16_t k, u
 
 int16_t VulkanAV1Decoder::Read_signed_primitive_refsubexpfin(uint16_t n, uint16_t k, int16_t ref)
 {
-    ref += n - 1;
-    const uint16_t scaled_n = (n << 1) - 1;
-    return Read_primitive_refsubexpfin(scaled_n, k, ref) - n + 1;
+    ref += (int16_t)(n - 1);
+    const uint16_t scaled_n = (uint16_t)(n << 1) - 1;
```

Comment:
Issues here are the number one, again. The number one needs to be cast to uint16_t.